### ISSUE TITLE
Difference Loss

### ISSUE THREAD

**Issue by DocF** [orion-orion/FedDCSR#1]
Could you tell me where your Difference loss is in this repo？

**Comment by orion-orion** [comment]
Hello~
The items of difference loss are computed in lines 219-222 and 224-238 of `train.py`, that is:
```python
        recons_loss = self.cs_criterion(
            result.reshape(-1, num_items + 1),
            ground.reshape(-1))  # (batch_size * seq_len, )
        recons_loss = (recons_loss * (ground_mask.reshape(-1))).mean()
```
and
```python
        kld_loss_s = -0.5 * \
            torch.sum(1 + logvar_s - mu_s ** 2 -
                      logvar_s.exp(), dim=-1).reshape(-1)
        kld_loss_s = (kld_loss_s * (ground_mask.reshape(-1))).mean()

        kld_loss_e = -0.5 * \
            torch.sum(1 + logvar_e - mu_e ** 2 -
                      logvar_e.exp(), dim=-1).reshape(-1)
        kld_loss_e = (kld_loss_e * (ground_mask.reshape(-1))).mean()
```
In lines 263-264 of `train.py`, we add them up to get the complete diffrence loss:
```python
        loss = alpha * (recons_loss + kld_weight * kld_loss_s + kld_weight
                        * kld_loss_e) \
```

**Comment by DocF** [comment]
thx

**Comment by orion-orion** [comment]
ヽ(*^ｰ^)人(^ｰ^*)ノ
